web/public/js/directives/event/cd-create-event/template.dust

```diff
@@ -119,16 +119,17 @@
           </div>
         </div>
       </div>
-      <div class="form-group form-inline full-width" ng-show="!eventInfo.status">
+      <div class="form-group form-inline full-width">
         <div class="checkbox checkbox-block">
-          <input type="checkbox" id="prefill-address" ng-model="eventInfo.prefillAddress" ng-click="prefillDojoAddress()" ng-init="prefillDojoAddress()">
+          <input type="checkbox" id="prefill-address" ng-model="eventInfo.useDojoAddress"
+            ng-click="prefillDojoAddress()" ng-disabled="pastEvent"/>
           <label for="prefill-address">{@i18n key="Use Dojo listing address"/}</label>
           <i uib-tooltip="{@i18n key="Use Dojos listing address. This will pre-fill the address fields."/}" tooltip-placement="top" tooltip-trigger="click" class="info-button fa fa-info-circle event-tooltip"></i>
         </div>
       </div>
       <div class="form-group" ng-class="{'has-error': eventForm.$submitted && eventForm.city.$error.required}">
         <label for="city">{@i18n key="City"/}*:</label>
-        <ui-select ng-model="eventInfo.city" ng-disabled="pastEvent" ng-readonly="eventInfo.prefillAddress" name="city" id="city" ng-change="updateLocalStorage('city', eventInfo.city); eventInfo.address = null" required>
+        <ui-select ng-model="eventInfo.city" ng-disabled="pastEvent || eventInfo.useDojoAddress" ng-readonly="eventInfo.useDojoAddress" name="city" id="city" ng-change="updateLocalStorage('city', eventInfo.city); eventInfo.address = null" required>
```

Comment:
The `ng-readonly` here is pointless as the `ng-disabled` will override it as it's a superset of restrictions. I'd be inclined to just get rid of the `ng-readonly` here.